src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Program.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Monitoring.UnitTestApp.Scenarios;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using System.CommandLine.Builder;
+using System.CommandLine.Parsing;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTestApp
+{
+    internal class Program
+    {
+        public static Task<int> Main(string[] args)
+        {
+            return new CommandLineBuilder()
+                .AddCommand(SpinWaitScenario.Command())
+                .UseDefaults()
+                .Build()
+                .InvokeAsync(args);
+        }
+
+        public static async Task<int> RunScenarioAsync(Func<ILogger, Task<int>> func, CancellationToken token)
+        {
+            // Create JSON console logger so that app can communicate with test host
+            // with strucutured responses.
+            using ServiceProvider hostServices = new ServiceCollection()
+                .AddLogging(builder =>
+                {
+                    builder.AddFilter(typeof(Program).FullName, LogLevel.Debug)
```

Comment:
It would be nice to push all the helper functions here to a separate class, instead of Program. I know it's very aspirational but maybe at some point we can unify and redist these test runners as one package across diag/dotnet-monitor repos.

Reply:
Done